### PR TITLE
CM-761 All reporting functionality has added into *discussions*

### LINKED ISSUE #761: Graphql API: Added report functional Discussion
Steps:
1. Added DiscussionReport (or whatever is more convenient for you) to ReportType
Screenshot 2021-08-12 at 15 02 45
2. Added discussionId as non-required into CreateReportInput ( see above )
3. Added flag-ReportFlag to Discussion Schema
Screenshot 2021-08-12 at 15 05 44
4. Added reports/flag fields to graphql `discussion/discussions` queries
Screenshot 2021-08-12 at 15 06 26
 the same as here
Screenshot 2021-08-12 at 15 07 03
5.  Added Hide/Show functional for Discussions

### PR DESCRIPTION
**Pull request by SobolevYuri**

### CODE REVIEW

File: packages/graphql/src/schema/Types/Discussion/Queries/GetDiscussions.query.ts [hunk 1]
```diff
@@ -13,8 +14,15 @@ export const GetDiscussionsQuery = queryField('discussions', {
   },
   resolve: (root, args) => {
     return prisma.discussion.findMany({
-        where: args.where as any,
-          ...args.paginate as any
+        where: {
+          flag: {
+            notIn: [
+              ReportFlag.Hidden
```
Comment (MeyerPV) [inline packages/graphql/src/schema/Types/Discussion/Queries/GetDiscussions.query.ts]: You should return discussions/discussionMessages/Proposals regardless of flag

File: packages/graphql/src/schema/Types/Reports/Extensions/ReportReportedDiscussion.extension.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { extendType } from 'nexus';
+import { prisma } from '@common/core';
+
+export const ReportReportedDiscussionExtension = extendType({
```
Comment (MeyerPV) [inline packages/graphql/src/schema/Types/Reports/Extensions/ReportReportedDiscussion.extension.ts:4]: ReportDiscussionExtension better

Comment (SobolevYuri) [reply packages/graphql/src/schema/Types/Reports/Extensions/ReportReportedDiscussion.extension.ts:4]: Screenshot 2021-08-25 at 21 07 55

Comment (SobolevYuri) [reply packages/graphql/src/schema/Types/Reports/Extensions/ReportReportedDiscussion.extension.ts:4]: @MeyerPV are u sure? I'll have to alter all the other similar extensions too then...

File: packages/graphql/src/schema/Types/Discussion/Queries/GetDiscussions.query.ts (hunk 1 above)
Comment (SobolevYuri) [reply packages/graphql/src/schema/Types/Discussion/Queries/GetDiscussions.query.ts]: @MeyerPV does it mean that i should remove the existing filtering logic for all these queries?

Comment (SobolevYuri) [reply packages/graphql/src/schema/Types/Discussion/Queries/GetDiscussions.query.ts]: i mean filtering by watching on this ReportFlag.Hidden value

**Review by MeyerPV — APPROVED** [review]

**Review by watchtower314 — APPROVED** [review]

**Pull request merged**